### PR TITLE
[EAGLE-2594] Basic tests and updates yaml for running in GH

### PR DESCRIPTION
**Pull request by marco-clarifai**
NOTE: Thumbnail tests are blocked by ingress rule change being promoted to prod: https://github.com/Clarifai/clarifai/pull/15630  <-- Done ✅ 

Tests secure data hosting using test account and fetches the files using multiple auth methods with `requests` lib

Note: Enabling linting on `./scripts` files  changed single quotes to double quotes
Tested in all environments (dev/staging/prod)

### CODE REVIEW

File: tests/client/test_secure_data_hosting.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    both_channels,
+    raise_on_failure,
+    wait_for_inputs_upload,
+)
+
+req_session = requests.Session()
+
+PAT = os.environ.get("CLARIFAI_PAT_KEY_SECURE_HOSTING")
+API_KEY = os.environ.get("CLARIFAI_API_KEY_SECURE_HOSTING")
+SESSION_TOKEN = os.environ.get("CLARIFAI_SESSION_TOKEN_SECURE_HOSTING")
```
Comment (yangshengBE) [inline tests/client/test_secure_data_hosting.py:19]: do we use session token?

Comment (marco-clarifai) [reply tests/client/test_secure_data_hosting.py:19]: for the app owner, yes (we also use this in the e2es)

Comment (marco-clarifai) [reply tests/client/test_secure_data_hosting.py:19]: Oh sorry, I disabled them when I was trying to debug the thumbnail issue. These should be re-enabled.

File: scripts/app_and_key_for_tests.py [hunk 2]
```diff
@@ -66,53 +69,63 @@ def create_app(env_name):
...
+    assert pat_id, f"Invalid response {data}"
 
     # This print needs to be present so we can read the value in CI.
     print(pat_id)
 
 
+def create_session_token():
+    session_token, _ = _login()
+    assert session_token, "Unable to generate session token"
+    print(session_token)
```
Comment (yangshengBE) [inline scripts/app_and_key_for_tests.py:126]: should remove?

File: scripts/app_and_key_for_tests.py [hunk 3]
```diff
@@ -66,53 +69,63 @@ def create_app(env_name):
...
+    assert len(data["keys"]) == 1, f"Invalid response {data}"
+    assert "id" in data["keys"][0], f"Invalid response {data}"
+    pat_id = data["keys"][0]["id"]
+    assert pat_id, f"Invalid response {data}"
 
     # This print needs to be present so we can read the value in CI.
     print(pat_id)
 
 
+def create_session_token():
```
Comment (yangshengBE) [inline scripts/app_and_key_for_tests.py:123]: do we need this function?

File: scripts/app_and_key_for_tests.py [hunk 4]
```diff
@@ -121,113 +134,126 @@ def delete(app_id):
...
-        print('''DESCRIPTION: Creates and delete applications and API keys
-ARGUMENTS:
---create-app [env_name]      ... Creates a new application.
---create-key [app_id]        ... Creates a new API key.
---create-pat                 ... Creates a new PAT key.
---delete-app [app_id]        ... Deletes an application (API keys that use it are deleted as well).
---create-workflow [api_key]  ... Creates a sample workflow to be used in int. tests.
---help                       ... This text.''')
+    elif arguments.create_session_token:
+        create_session_token()
```
Comment (yangshengBE) [inline scripts/app_and_key_for_tests.py:214]: what do we do with session token created by this ?

**Review by yangshengBE — APPROVED** [review]

**Pull request merged**